### PR TITLE
Add translation of memory model from module metadata

### PR DESCRIPTION
**Pull request by NikitaRudenkoIntel**

### CODE REVIEW

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -105,6 +105,18 @@ static void foreachKernelArgMD(
   }
 }
 
+static SPIRVMemoryModelKind getMemoryModel(Module &M) {
+  NamedMDNode *MemoryModelMD = M.getNamedMetadata(kSPIRVMD::MemoryModel);
+  if (MemoryModelMD) {
+    auto Model = static_cast<SPIRVMemoryModelKind>(
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVWriter.cpp:111]: I think we should either return result of static cast directly, or break down this expression into several simpler ones, possibly with checks that pointers are not null (static analysis tools may find issues here).

Comment (NikitaRudenkoIntel) [reply lib/SPIRV/SPIRVWriter.cpp]: I added some checks

File: test/memory_model_md.ll [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+; RUN: llvm-dis %t.bc -o %t.ll
+; RUN: FileCheck %s --input-file %t.ll  -check-prefix=LLVM
+; RUN: FileCheck %s --input-file %t.spt -check-prefix=SPV
+
+; ModuleID = 'float_control_empty.bc'
+source_filename = "float_control_empty.cpp"
+target datalayout = "e-p:64:64-i64:64-n8:16:32"
+target triple = "spir"
+
+; LLVM-DAG: !spirv.MemoryModel = !{![[MMMD:[0-9]+]]}
```
Comment (svenvh) [inline test/memory_model_md.ll:14]: The use of `-DAG` feels a bit too strong for this case.

Comment (NikitaRudenkoIntel) [reply test/memory_model_md.ll]: Fixed

**Pull request merged**